lib/core/build/classes/datawrapper/Chart.php

```diff
@@ -144,70 +146,46 @@ protected function getDataFilename() {
     }
 
     /**
-     * writes any asset to the file system store
+     * writes any asset to the asset store using v3 api
      */
     public function writeAsset($filename, $data) {
-        $cfg = $GLOBALS['dw_config'];
+        if (!$this->assets) $this->assets = [];
 
-        if (isset($cfg['charts-s3'])
-          && isset($cfg['charts-s3']['write'])
-          && $cfg['charts-s3']['write'] == true) {
+        [$status, $body] = call_v3_api('PUT',
```

Comment:
I think we should check the `$status` of the v3 api request and throw some sort of error if the request failed.

Reply:
yes, makes sense